### PR TITLE
Add Ktlint

### PR DESCRIPTION
**Pull request by nplasterer**
Per @elisealix22 suggested here: https://github.com/xmtp/xmtp-android/pull/5#pullrequestreview-1245739323
Per documentation here: https://pinterest.github.io/ktlint/install/integrations/

This adds a new linter task to our CI and lints the existing code.

### CODE REVIEW

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -22,3 +22,28 @@ jobs:
         uses: gradle/wrapper-validation-action@v1
       - name: Gradle Run Tests
         run: ./gradlew library:testDebug
+  static-analysis:
```
Comment (nplasterer) [inline .github/workflows/test.yml:25]: I could be down to moving this into a new workflow file called `lint.yml` instead of adding it to test. Thoughts?

Comment (elisealix22) [reply .github/workflows/test.yml:25]: Up to you IMO! Both work for setting up the gh rule I believe so just personal pref

**Review by elisealix22 — APPROVED** [review]
💚

File: .github/workflows/test.yml (hunk 1 above)
Comment (nplasterer) [reply .github/workflows/test.yml]: I decided to move it. Since we will add the example app tests in this file as well.

**Pull request merged**